### DISCUSSION TITLE
What is the purpose of keyPrefix?

### DISCUSSION THREAD

**Discussion by sadeghhosseini** [animir/node-rate-limiter-flexible#211] [Q&A] [answered]
[Here](https://github.com/animir/node-rate-limiter-flexible/wiki/Options#keyprefix) it is stated:
> If you need to create several limiters for different purpose.

What does it mean by "different purpose"?

**Comment by animir** [comment, marked as answer]
@sadeghhosseini 
It is required when you need to create two or more limiters with different options so keys don't interfere with different limiters.
I changed that description, thank you for the question.

**Reply by sadeghhosseini** [reply]
Thanks for taking the time to respond. But it is still a little bit ambiguous to me.

Consider the following `express` example

```javascript
const express = require('express')
const { RateLimiterMemory } = require('rate-limiter-flexible');
const app = express()
const port = 3000

const rateLimiter1 = new RateLimiterMemory({
    points: 1,
    duration: 60,
});
const rateLimiter2 = new RateLimiterMemory({
    points: 10,
    duration: 60,
});
const rateLimiterMiddleware1 = (req, res, next) => {
    rateLimiter1.consume(req.ip)
        .then(() => {
            next();
        })
        .catch(() => {
            res.status(429).send('Too Many Requests');
        });
};

const rateLimiterMiddleware2 = (req, res, next) => {
    rateLimiter2.consume(req.ip)
        .then(() => {
            next();
        })
        .catch(() => {
            res.status(429).send('Too Many Requests');
        });
};
app.get('/endpoint1', rateLimiterMiddleware1, (req, res) => {
    res.send('Hello World!')
})
app.get('/endpoint2', rateLimiterMiddleware2, (req, res) => {
    res.send('Hello World!')
})

app.listen(port, () => {
    console.log(`Example app listening on port ${port}`)
})
```
`rateLimiter1` and `rateLimiter2` are two limiters with different options and both have the same key `req.ip`. As I tested if a user with IP x.x.x.x hit `/endpoint1`  twice they will be ratelimited and then if they hit `/endpoint2` right after that they won't get ratelimited, they should hit the endpoint 11 times to get ratelimited as expected, which means keys don't seem to interfere.

Because of that I think by "create two or more limiters" you mean, creating them in a way other than the way done in the preceding example.

**Reply by animir** [reply]
This is a good example. The memory limiter is not like any other store limiter from this package.
It stores keys in separate objects for every limiter, so you don't face any issues.
